autobahn/websocket/types.py

```diff
@@ -127,6 +127,96 @@ def __str__(self):
         return json.dumps(self.__json__())
 
 
+@public
+class ConnectingRequest(object):
+    """
+    Thin-wrapper for WebSocket connection request information provided in
+    :meth:`autobahn.websocket.protocol.WebSocketClientProtocol.onConnecting`
+    after a client has connected, but before the handshake has
+    proceeded.
+
+    `host`, `port`, and `resource` are all required, everything else
+    is optional. Note that these are values that will be seen by the
+    client and should represent the public-facing host, port and
+    resource to which the client is connecting (not necessarily the
+    action host/port being used).
+    """
+
+    __slots__ = (
+        'host',
+        'port',
+        'resource',
+        'headers',
+        'useragent',
+        'origin',
+        'protocols',
+    )
+
+    def __init__(self, host, port, resource, headers=None, useragent=None, origin=None, protocols=None):
```

Comment:
With WebSocket over non-IP based transports (such as UDS, pipes, serial connections) there is no host or port at the bytes transport level. WebSocket itself however requires a fully qualified HTTP URL - and hence host+port are "needed" in as for creating the HTTP URL for the initial HTTP request that start the WebSocket opening handshake. Eg WebSocket-over-UDS, I often simply use `http://localhost` (because it really doesn't matter .. it's just to make WebSocket happy).

Because of that, I'm not sure if we want to require host and port in the ctor here .. we might make it optional, and fill in defaults automatically (host=="localhost", port==80, resource=="/")

Reply:
Well, I think we do: I simply copied everything we used from `WebSocketClientFactory` -- and `host, port, resource` were all used without checking. The others might have been null. My reading of the code is that these are all "whatever is public"...? Or something? e.g. here is the original code: https://github.com/crossbario/autobahn-python/blob/master/autobahn/websocket/protocol.py#L3378

Reply:
Oh, wait, I see what you mean, maybe: if they're `None` in ctor then we just make something up (presuming the caller means "I don't care")...? (where "something" == "ws://localhost:80/")

Reply:
Oh, yeah, that's exactly what the old code would have done: https://github.com/crossbario/autobahn-python/blob/master/autobahn/websocket/protocol.py#L3838